### PR TITLE
fix broadcast piracy

### PR DESCRIPTION
**Pull request by KirillZubov**
https://github.com/SciML/NeuralPDE.jl/issues/337

### CODE REVIEW

File: src/pinns_pde_solve.jl [hunk 1]
```diff
@@ -185,6 +185,9 @@ where
 function _transform_expression(ex,dict_indvars,dict_depvars,chain,eltypeθ,strategy)
     _args = ex.args
     for (i,e) in enumerate(_args)
+        if e in [+,-,*,/,^,inv]
```
Comment (ChrisRackauckas) [inline src/pinns_pde_solve.jl:188]: why is this necessary?

Comment (ashutosh-b-b) [reply src/pinns_pde_solve.jl:188]: Inorder to disable `__dot__` for specific functions.

Comment (ChrisRackauckas) [reply src/pinns_pde_solve.jl:188]: but why these functions, and why here? I don't get it.

Comment (KirillZubov) [reply src/pinns_pde_solve.jl:188]: sorry for the long answer, 
the problem is that `toexpr` transforms not as symbols but as functions of all arithmetic operators. But in Broadcast.jl the function is used as a marker for not dotting (which is generally weird)
https://github.com/JuliaLang/julia/blob/4270d3b44e53e9fa68dcdbb6cb416d232615db4c/base/broadcast.jl#L1295


```julia
@parameters x y
@variables u(..)
eq =  u(x)+ u(x)*x
expr = toexpr(eq)
julia> dump(expr)
Expr
  head: Symbol call
  args: Array{Any}((3,))
    1: + (function of type typeof(+))
    ...

Base.Broadcast.dottable(x::Function) = false
julia> Broadcast.__dot__(expr)
:((+)(u.(x), (*)(x, u.(x))))

Base.Broadcast.dottable(x::Function) = true
julia> Broadcast.__dot__(expr)
:((+).(u.(x), (*).(x, u.(x))))
```

Comment (KirillZubov) [reply src/pinns_pde_solve.jl:188]: so to avoid piracy  and broadcasted full expression with arithmetic in need convert them in symbol

Comment (KirillZubov) [reply src/pinns_pde_solve.jl:188]: in general, this is a crutch, using functions inside an expression to avoid dotting. 
https://github.com/JuliaLang/julia/blob/4270d3b44e53e9fa68dcdbb6cb416d232615db4c/base/broadcast.jl#L1295

Comment (KirillZubov) [reply src/pinns_pde_solve.jl:188]: leading to the fact that we have to exclude functions in order to correct broadcast expr, which is kind of bad

Comment (ChrisRackauckas) [reply src/pinns_pde_solve.jl:188]:
```suggestion
        if e isa Function
```

?

Comment (KirillZubov) [reply src/pinns_pde_solve.jl:188]: it doesn't work,   `Differential` is also function( .
```julia
Dθ = Differential(θ)
eq = Dθ(u(θ))
expr = toexpr(eq)
julia> dump(expr)
Expr
  head: Symbol call
  args: Array{Any}((2,))
    1: Differential(θ) (function of type Differential)
...
```

Comment (KirillZubov) [reply src/pinns_pde_solve.jl:188]: to avoid all this, in the end, we need our own `toexpr` in NeuralPDE

Comment (ChrisRackauckas) [reply src/pinns_pde_solve.jl:188]: `if e isa Function && e !== Differential`

Comment (ChrisRackauckas) [reply src/pinns_pde_solve.jl:188]: >to avoid all this, in the end, we need our own toexpr in NeuralPDE

That seems unnecessary.

Comment (ChrisRackauckas) [reply src/pinns_pde_solve.jl:188]: Why would it require an entire new `toexpr` instead of just omitting one function?

Comment (KirillZubov) [reply src/pinns_pde_solve.jl:188]: Since `ModelingToolKit.toexpr` does not guarantee the same behavior, many times it was necessary to rewrite the parser after updating ModelingToolKit.jl

but it’s not urgent, as long as it works)

**Review by ChrisRackauckas — APPROVED** [review]
